.github/workflows/integration-tests.yml

```diff
@@ -69,9 +69,20 @@ jobs:
       - name: Run integration tests - 1 peer
         run: poetry -C tests run ./tests/integration-tests.sh distributed
         shell: bash
+      - name: Restore pytest cache
+        uses: actions/cache@v3
+        with:
+          path: .pytest_cache
+          key: pytest-cache-${{ github.run_id }}
+          restore-keys: pytest-cache-
```

Comment:
In [CI documentation](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/caching-dependencies-to-speed-up-workflows#cache-hits-and-misses) I read this:

> `restore-keys`: (...)  
If there are no exact matches, the action searches for partial matches of the restore keys. When the action finds a partial match, the most recent cache is restored to the path directory.

Would it make sense to also include the current branch name in this cache key? That may prevent taking caches from other branches/PRs that are unrelated.